src/main/java/ch/heigvd/dai/tcp/ClientParser.java

```diff
@@ -0,0 +1,226 @@
+        break;
+      }
+    }
+
+    int status = Integer.parseInt(buffer.toString().trim());
+    if (status != 0) {
+      System.err.println("Got error: " + Errno.getErrorMessage(status));
+
+      return false;
+    }
+
+    return true;
+  }
+
+  private void list(String path) throws IOException {
+    String command = "LIST " + path + Server.NEW_LINE;
+    sendRequest(command);
+
+    if (!parseStatus()) {
+      return;
+    }
+
+    StringBuilder result = new StringBuilder();
+
+    int byteRead;
+    while ((byteRead = in.read()) != -1) {
+      char c = (char) byteRead;
+      if (c == Server.EOT) {
+        break;
+      }
+      result.append(c);
+    }
+
+    System.out.println("Got result: ");
+    Arrays.stream(result.toString().split(Server.DELIMITER)).forEach(s -> {
+      System.out.println(s);
+    });
+  }
+
+  public void delete(String path) throws IOException {
+    String command = "DELETE " + path + Server.NEW_LINE;
+    sendRequest(command);
+
+    if (!parseStatus()) {
+      return;
+    }
+
+    System.out.println("Target deleted successfully");
+  }
+
+  /**
+   * Upload a file to the server
+   */
+  private void put(Path localPath, String remotePath) throws IOException {
+    File file = localPath.toFile();
+
+    if (!file.exists()) {
+      System.err.println("Local file doesn't exist");
+      return;
+    }
+
+    if (!file.isFile()) {
+      System.err.println("Path doesn't point to a file");
+      return;
+    }
+
+    if (remotePath.endsWith("/")) {
+      remotePath += file.getName();
+    }
+
+    try (FileInputStream fin = new FileInputStream(file);) {
+      System.out.println("Sending file");
+      sendRequest("PUT " + remotePath + " " + file.length() + "\n");
+
+      // read the file using 4K blocks
+      byte[] buffer = new byte[4096];
+      int bytesRead;
+
+      while ((bytesRead = fin.read(buffer)) != -1) {
+        out.write(buffer, 0, bytesRead);
+      }
+      out.flush();
+      System.out.println("file sent");
+    } catch (FileNotFoundException e) {
+      System.err.println("Unable to open file");
+    }
+
+    // Check if the server reveived correctly
+
+    System.out.println("waiting for answer");
+    if (!parseStatus()) {
+      return;
+    }
+
+    System.out.println("Uploaded successfully");
+  }
+
+  /**
+   * Create a directory on the server
+   */
+  private void mkdir(String remotePath) throws IOException {
+
+    System.out.println("Sending PUT request to create directory");
+    if (!remotePath.endsWith("/")) {
+      remotePath = remotePath + "/";
+    }
+
+    String command = "PUT " + remotePath + Server.NEW_LINE;
+    sendRequest(command);
+
+    if (!parseStatus()) {
+      return;
+    }
+
+    System.out.println("directory created successfully");
+  }
+
+  @Override
+  public void parse(String[] tokens) throws IOException, ServerHasGoneException {
+    super.parse(tokens);
+
+    switch (tokens[0]) {
+      case "LIST", "ls", "list" -> {
```

Comment:
The original goal was to provide aliases for the client REPL which would be better for UX.
We could use TOUPPER to reduce the number of checks. At the time I just thought it made sense that way since I wouldn't expect the user to alternate cases except of List maybe.

We can fix this later on once most of the requirements from the evaluation grid are taken care of.